core/types/deposit_tx_kroma.go

```diff
@@ -0,0 +1,112 @@
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package types
+
+import (
+	"bytes"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type KromaDepositTx struct {
+	// SourceHash uniquely identifies the source of the deposit
+	SourceHash common.Hash
+	// From is exposed through the types.Signer, not through TxData
+	From common.Address
+	// nil means contract creation
+	To *common.Address `rlp:"nil"`
+	// Mint is minted on L2, locked on L1, nil if no minting.
+	Mint *big.Int `rlp:"nil"`
+	// Value is transferred from L2 balance, executed after Mint (if any)
+	Value *big.Int
+	// gas limit
+	Gas uint64
+	// Normal Tx data
+	Data []byte
+}
+
+// copy creates a deep copy of the transaction data and initializes all fields.
+func (tx *KromaDepositTx) copy() TxData {
+	cpy := &KromaDepositTx{
+		SourceHash: tx.SourceHash,
+		From:       tx.From,
+		To:         copyAddressPtr(tx.To),
+		Mint:       nil,
+		Value:      new(big.Int),
+		Gas:        tx.Gas,
+		Data:       common.CopyBytes(tx.Data),
+	}
+	if tx.Mint != nil {
+		cpy.Mint = new(big.Int).Set(tx.Mint)
+	}
+	if tx.Value != nil {
+		cpy.Value.Set(tx.Value)
+	}
+	return cpy
+}
+
+// accessors for innerTx.
+func (tx *KromaDepositTx) txType() byte           { return DepositTxType }
+func (tx *KromaDepositTx) chainID() *big.Int      { return common.Big0 }
+func (tx *KromaDepositTx) accessList() AccessList { return nil }
+func (tx *KromaDepositTx) data() []byte           { return tx.Data }
+func (tx *KromaDepositTx) gas() uint64            { return tx.Gas }
+func (tx *KromaDepositTx) gasFeeCap() *big.Int    { return new(big.Int) }
+func (tx *KromaDepositTx) gasTipCap() *big.Int    { return new(big.Int) }
+func (tx *KromaDepositTx) gasPrice() *big.Int     { return new(big.Int) }
+func (tx *KromaDepositTx) value() *big.Int        { return tx.Value }
+func (tx *KromaDepositTx) nonce() uint64          { return 0 }
+func (tx *KromaDepositTx) to() *common.Address    { return tx.To }
+func (tx *KromaDepositTx) isSystemTx() bool       { return false }
+
+func (tx *KromaDepositTx) effectiveGasPrice(dst *big.Int, baseFee *big.Int) *big.Int {
+	return dst.Set(new(big.Int))
+}
+
+func (tx *KromaDepositTx) effectiveNonce() *uint64 { return nil }
+
+func (tx *KromaDepositTx) rawSignatureValues() (v, r, s *big.Int) {
+	return common.Big0, common.Big0, common.Big0
+}
+
+func (tx *KromaDepositTx) setSignatureValues(chainID, v, r, s *big.Int) {
+	// this is a noop for deposit transactions
+}
+
+func (tx *KromaDepositTx) encode(b *bytes.Buffer) error {
+	return rlp.Encode(b, tx)
+}
+
+func (tx *KromaDepositTx) decode(input []byte) error {
+	return rlp.DecodeBytes(input, tx)
+}
+
+func IsKromaDepositTx(input []byte) bool {
+	var cnt int
+	buf, _, _ := rlp.SplitList(input)
+	for len(buf) > 0 {
+		var err error
+		_, buf, err = rlp.SplitString(buf)
+		if err != nil {
+			panic(err)
+		}
+		cnt++
+	}
```

Comment:
I think this will work the same way
```suggestion
	buf, _, _ := rlp.SplitList(input)
	cnt, _ := rlp.CountValues(buf)
```